### PR TITLE
Clang tidy

### PR DESCRIPTION
**Pull request by rpavlik**
A variety of cleanups prompted by running clang-tidy (and include-what-you-use) over the code in src. Should be mostly simple/small things - no major refactoring this time.

### CODE REVIEW

File: src/api_layers/api_dump.cpp [hunk 1]
```diff
@@ -393,7 +399,7 @@ bool ApiDumpLayerRecordContent(std::vector<std::tuple<std::string, std::string,
     return success;
 }
 
-XrResult ApiDumpLayerXrCreateInstance(const XrInstanceCreateInfo *info, XrInstance *instance) {
+XrResult ApiDumpLayerXrCreateInstance(const XrInstanceCreateInfo * /*info*/, XrInstance * /*instance*/) {
```
Comment (daveh-lunarg) [inline src/api_layers/api_dump.cpp:402]: Actually surprised this even compiles.  But, if the params are unused, should just remove them - right?

File: src/api_layers/api_dump.cpp [hunk 2]
```diff
@@ -589,7 +591,7 @@ extern "C" {
 
 // Function used to negotiate an interface betewen the loader and an API layer.  Each library exposing one or
 // more API layers needs to expose at least this function.
-LAYER_EXPORT XrResult xrNegotiateLoaderApiLayerInterface(const XrNegotiateLoaderInfo *loaderInfo, const char *apiLayerName,
+LAYER_EXPORT XrResult xrNegotiateLoaderApiLayerInterface(const XrNegotiateLoaderInfo *loaderInfo, const char * /*apiLayerName*/,
```
Comment (daveh-lunarg) [inline src/api_layers/api_dump.cpp]: Same comment, just remove unused ?

File: src/api_layers/api_dump.cpp (hunk 1 above)
Comment (daveh-lunarg) [reply src/api_layers/api_dump.cpp:402]: Nevermind - this is shorthand for 'unused param', and signature cant change.  Learned something new!

**Review by daveh-lunarg — APPROVED** [review]
Not previously familiar with clang-tidy, but I do like the tidier version.

Except the brace locations - but that's google style clang-format's fault :(

File: src/api_layers/api_dump.cpp (hunk 1 above)
Comment (rpavlik) [reply src/api_layers/api_dump.cpp:402]: yup yup :) It's like putting `_` at the beginning of a param/variable name in a variety of other languages (python, rust) - basically meaning "I know this isn't used, but I can't/won't change the signature so don't complain about it, compiler/linter."

**Comment by rpavlik** [comment]
Clang tidy has definitely gotten more powerful since I had last applied its "fix" mode - what with entirely removing unused parameters for purely-internal (file-static) functions, really smart range-for conversion, etc. I was quite impressed - and quite bummed I had done some of those cleanups myself manually.

**Comment by rpavlik** [comment]
I'll merge this after it finishes being reviewed in the internal GitLab.

**Comment by rpavlik** [comment]
Incorporated in GitLab and will be pushed with next sync.

**Pull request closed without merging**